client/src/main/java/io/pravega/client/stream/ReaderGroupConfig.java

```diff
@@ -49,11 +50,46 @@
 
     private final int maxOutstandingCheckpointRequest;
 
+    private final StreamDataRetention retentionConfig;
+
+    /**
+     * If a Stream's Retention Policy {@link RetentionPolicy} is set to
+     * 'CONSUMPTION' based retention {@link RetentionPolicy.RetentionType#CONSUMPTION} ,
+     * the Reader Group needs to periodically notify StreamCuts to Controller to indicate
+     * the point in the Stream till which it has completed consuming data.
+     * This can be done by setting the retentionConfig in ReaderGroupConfig {@link ReaderGroupConfig}
+     * to 'CONSUMPTION_BASED_USER_STREAMCUT' or 'CONSUMPTION_BASED_AT_LAST_CHECKPOINT'.
+     * If a Stream's Retention Policy is not Consumption based the ReaderGroupConfig should have retentionConfig = 'NO_IMPACT'.
+     *
+     * Note: It is incorrect to set ReaderGroup retentionConfig = 'NO_IMPACT' when
+     * a Streams' Retention Policy = 'CONSUMPTION' {@link RetentionPolicy.RetentionType#CONSUMPTION},
+     * because with NO_IMPACT the ReaderGroup will not notify consumption Stream-Cuts to Controller
+     * and so Consumption based retention of data in the Stream would not happen.
```

Comment:
```suggestion
```
A stream may have multiple reader groups, where one wishes to retain the data explicitly and the other does not care. This is an important usecase.

Reply:
I have stated both in the javadoc now.